adding bot translator sample

amr-elsehemy opened the pull request.

enzocano committed:
Add Enterprise Bot Template

Squashed commit:

[37dacb50] Fix typo

[f3ac88d9] Review documentation

[3dc4545e] Add (c) header

[5a377e62] Delete unused classes

[74de8edf] Formatting code

[85835639] Build before start bot

[6d61c025] Change string keys to env.vars

[01a77427] Add bot section to bot.recipe for successful msbot clone services command

[f3ba106c] Add env.development vars

[a35f20c7] Fix LUIS appName

[48464a53] Fix issue with LUIS top intent

[e18ddd86] Migrate documentation and apply minor fixes

[c6677bb0] Various fixes

[1b42ae5a] Padding tsconfig

[dcd46bb7]  Fix resources parsing

[9d8c6f52] Fix propAccessor initialization

[35752d78] Add .map files for easy debug

[11970588] Use single comma

[67fb82a4] Fix resources name

[8c832528] Delete unused resources

[7a8c2807] Add clean and copy-resources scripts

[af5b051a] Implement onboardingDialog.ts and onboardingResponses.ts

[2fdf37e9] Enable use of 'CancelResponses' in 'enterpriseDialog'

[59c40c5f] Fix wrong parameter in 'signInResponses'

[6ca779a8] implemented escalateDialog.ts

[0b06789e] implemented escalateState.ts

[b27c8608] Fix MainDialog properties

[e19c4001] implemented escalateResponses.ts

[65c8994a] implemented cancelState.ts + minor typo fix

[45d7df45] Implement EnterpriseBotAccessors

[04ff4245] Implement cancelResponses.ts

[ed48b4c0] Implement GraphClient

[1bbf84fd] Add SignInDialog

[711d5715] Reorganize imports and remove unused imports

[090d842d] Fix mainDialog

[2acdb8ef] Add files for 'dialogs/shared/'
Add the following files to 'dialogs/shared/':
- interruptableStatus.ts
- interruptableDialog.ts
- enterpriseDialog.ts

[34b2d2ed] Add auto-generated code from LuisGen

[67d6cfbd] Implement mainResponses

[cb07ec2d] Add ActivityEx.createReply

[6308cb4c] Add resourceParser

[9fdc1e7f] Add enterpriseBot

[c3bce667] Fix telemetry send messages

[0cac042d] Add contentModerator middleware

[74a27984] Add luis and qna telemetry

[0cc5ec80] Add TelemetryLoggerMiddleware implementation

[cbe8ccac] Rebase with master

JasonSowers committed:
fix broken link in qna samples

JasonSowers committed:
Update README.md

JasonSowers committed:
Update README.md

stevengum committed:
Merge pull request #573 from Microsoft/JasonSower-qna-readme-fix

Update broken link in readme

stevengum committed:
Merge pull request #575 from Microsoft/JasonSowers-link2

fix broken link in qna samples

daveta committed:
Address feedback

vishwacsena committed:
Fix for #436 - VSIX sources. (#572)

daveta committed:
Merge pull request #577 from Microsoft/daveta-cafebot3

Cafebot ASP.Net Core

Commit:
adding sample

Commit:
adding auth bot

vishwacsena committed:
Fix for #436 in samples. (#571)

daveta committed:
Fix tons of issues in here. (#576)

JasonSowers committed:
Fix node/12.nlp-with-luis (#581)

* fixed the bot

* Update README.MD

JasonSowers committed:
Correct package.json

stevengum committed:
change '\' to '/' clean up paths and other errors in .netcore READMEs (#592)

* change '\' to '/' clean up paths and other errors in .netcore READMEs

* additional cleanup on csharp_dotnetcore READMEs

* update .netcore samples 09, 13 READMEs

stevengum committed:
Replace "\" with "/" in JS READMEs and cleanup es6 READMEs (#588)

* update nodejs samples' READMEs

* update ts 02 sample for unix-compat instructions

* cleanup js_es6 READMEs

JasonSowers committed:
Update readme for c#/12.nlp-with-luis (#583)

* Update README.md

* Update README.md

* Update README.md

stevengum committed:
add Installing_CLI_tools.md (#587)

vishwacsena committed:
adding link to c# cafe bot (#589)

JasonSowers committed:
Edit readme for node/12.nlp-with-luis (#584)

Making the same as the C# PR #583

Commit:
Connection setting variable was undefined when signing out (#591)

benbrown committed:
Add eslint configuration

daveta committed:
Fix missing default.htm and accomodate any underscores in bot name for LUIS/Dispatch

JasonSowers committed:
Merge pull request #582 from Microsoft/jassow/18.bot-auth

Jassow/18.bot auth

worzy committed:
Fix filename in logged output when running nodejs sample 14 (#597)

benbrown committed:
Correct formatting

JasonSowers committed:
Merge pull request #599 from Microsoft/benbrown-patch-1

Correct formatting

JasonSowers committed:
Merge pull request #598 from Microsoft/daveta-dispatch2

Fix missing default.htm and accommodate any underscores in bot name

enzocano committed:
Update main readme

enzocano committed:
Fix tslint issues

Commit:
Fix random number generation sample 7

Commit:
[Yeoman] TypeScript Basic template prompt bug (#603)

* add package.version in various places

- Display package version when the generator is run
- Write the package version out in the generated package.json files
- bump the version number of the generator

* Fixes 585 - [Yeoman] TypeScript Basic template prompt bug

- TypeScript generated code was missing a break statement.
- Also updated the inline comments to be consistent with the JavaScript version
- fix yeoman generated code
- fix JavaScript sample
- fix TypeScript sample

Commit:
Fixes 556 - Subscription key in C# #14 readme #556 (#604)

- remove subscription key
- fix markdown formatting

enzocano committed:
Fix DEPLOYMENT.md link

JasonSowers committed:
Merge pull request #602 from Microsoft/kydela

Fix random number generation sample 7

enzocano committed:
Change ContentModerator config key snippet to match bot.recipe

enzocano committed:
Check QnA service name

stevengum committed:
update .netcore Dispatch sample to get into a working state

stevengum committed:
update .netcore Dispatch sample to get into a working state

stevengum committed:
merge master into stgum/csharp_dispatch

stevengum committed:
merge master into stgum/csharp_dispatch

jcscottiii committed:
Remove repeating word in README

This commit cleans up the README by removing the repeating, successive words

enzocano committed:
Delete unused code

VictorGrycuk committed:
Add update step for VS Code, improve format

joseques committed:
Fixed minor typos for README

stevengum committed:
Merge pull request #614 from jcscottiii/patch-1

Remove repeating word in README

stevengum committed:
fix v4 Node.js graph sample

JasonSowers committed:
Merge pull request #621 from Microsoft/stgum/node-graph-fix

fix v4 Node.js graph sample

dfavretto committed:
Readme fixes (#610)

alermant committed:
Fix broken links (#617)

daveta committed:
Fix readme commands for msbot clone (#586)

* Fix readme commands for msbot clone

* Fixup more references to js

* Use fully designated forms of command line parameter so more self describing

JasonSowers committed:
update c#/24.bot-authentication-msgraph (#612)

* fixed sample

* cleanup

lauren-mills committed:
Merge pull request #608 from southworkscom/southworks/netcore/fix-52.enterprise-bot

[netcore][52.enterprise bot] Fix ContentModerator service key and QnA service name

lauren-mills committed:
Merge pull request #615 from southworkscom/southworks/netcore/clean-52.enterprise-bot

[netcore][52.enterprise bot] Delete unused code

mr-yamraj committed:
typo those to that

JasonSowers committed:
Merge pull request #626 from mr-yamraj/master

typo those to that

Commit:
fix: fix unreachable code and potential uncaught exception (#632)

talon2king committed:
Fixes to CafeBots OnTurnProperty and BookTableDialog classes

enzocano committed:
FIx typos

VictorGrycuk committed:
Rephrase 'Hit F5'

enzocano committed:
Fix typos

enzocano committed:
Use nodejs slash

stevengum committed:
Merge pull request #642 from southworkscom/southworks/netcore/console-echo-readme

[netcore][01.console echo] Add missing step and improve format in Readme

stevengum committed:
Merge pull request #641 from southworkscom/southworks/nodejs/typo-15.handling-attachments

[nodejs][15.handling-attachments] Fix typos

stevengum committed:
Merge pull request #640 from southworkscom/southworks/typescript/console-echo

[typescript][01.console echo] Fix minor typos for README

stevengum committed:
Merge pull request #638 from southworkscom/southworks/nodejs/typo-14.nlp-with-dispatch

[nodejs][14.nlp-with-dispatch] Fix typos

talon2king committed:
Cleaned up code and removed comments, todos and old code

luisprieto committed:
remove unused requiring modules in nodejs message-routing sample

enzocano committed:
Add proactive message concept to readme

enzocano committed:
Add "show" reference option

joseques committed:
Fix typos and changes in README

enzocano committed:
Change .bot reference

stevengum committed:
Merge pull request #650 from southworkscom/southworks/nodejs/fix-17.multilingual-conversations

[nodejs][17.multilingual-conversations] Edit .bot reference

stevengum committed:
Merge pull request #647 from southworkscom/southworks/nodejs/echo-w-counter

[nodejs][02.echo with counter] Fixed minor typos for README

stevengum committed:
Merge pull request #646 from southworkscom/southworks/nodejs/fix-16.proactive-messages

[nodejs][16.proactive-messages] Add documentation about proactive messages

stevengum committed:
Merge pull request #644 from luisprieto/master

remove unused requiring modules in nodejs message-routing sample

Commit:
resized images to work on all channels

daveta committed:
Merge pull request #652 from Microsoft/jassow-image

c#/node Attachments sample resized images to work on channels

lauren-mills committed:
Merge pull request #601 from southworkscom/southworks/typescript/enterprise

[TypeScript] Enterprise Bot

daveta committed:
Bit updated on readme, pull from appsettings correctly

halilb committed:
fixing watch script for typescript samples

daveta committed:
Relax validation on Echobot, tighten screws on BasicBot

vishwacsena committed:
Cafebot nodejs sample cleanup (#633)

* cleanup cafe bot.

* more fixes. trimming to one luis call per turn.

* more cleanup

* More fixes.

* eslint fix.

* recipe file fixes.

* clean up.

* Updates to cafe bot.

enzocano committed:
[nodejs][19.custom-dialogs] Fix dependencies & readme.md (#651)

* Delete unused dependencies

* Add nodemon run

* Use nodejs-style file names

Commit:
[netcore][11.qnamaker] Fix typos and improve formatting (#656)

* Edit README.md

* Fix typos

* Remove extra space from comment

enzocano committed:
[nodejs][21.luis-with-appinsights] Fix luis configuration & typos (#666)

* Use luisConfig.getEndpoint()

* Use authoringKey if not subscriptionKey

* Fix typos

* Change endpointKey to authoringKey

Commit:
[netcore][07.adaptive cards] Edit Readme (#611)

* Edit readme.md

* Fix typos in code comments

* Remove Azure storage section as per feedback

* Edit deployment step

Commit:
[typescript][02.echo with counter] Fix typos and improve structure in Readme (#663)

* Fixed minor typos for README

* Changed 'Deploy to Azure' section

Commit:
[netcore][02.echo with counter] Fix typos and improve Readme.md (#672)

* Fix typos and update Readme.md

* Apply feedback corrections

Commit:
[netcore][13.basic bot] Fix typos and improve formatting (#671)

* Fix file name reference, deploy section, and style

* Fix typos in code and minor style edits

* Fix typo

Commit:
[netcore][06.using cards] Fix typos and update Readme structure and content (#670)

* Fix typos in comments

* Edit Readme
- Add Testing section
- Improve formatting
- Reorder min req for tool version

* Fixes to Readme.md

Commit:
[typescript][13.basic bot] Fix typo and add missing link in Readme (#667)

* Fixed minor typos for README.md and DEPLOY.md

* Minor typo in bot.ts

nishanc committed:
Update to bot authorization link.

Previous link points to SDK v3 release related article.

amr-elsehemy committed:
Adding bot translator sample

amr-elsehemy committed:
Merge branch 'master' of https://github.com/Microsoft/BotBuilder-Samples

daveta committed:
Merge pull request #676 from nishanc/patch-1

Update to bot authorization link.

joseques committed:
Minor fixes and typos

Commit:
[netcore][15.handling attachments] Fix typos and improve formatting (#680)

* Edit readme.md

* Fix typos and minor format

* Add missing steps for BFM V4, and deploy

* Fix typos in attachments section

* Fix readme structure

enzocano committed:
Update readme (#679)

enzocano committed:
[nodejs][24.bot-authentication-msgraph] Fix readme.md (#677)

* Order required permissions

* Fix auth documentation link

Commit:
[netcore][09.message routing] Fix typos and text format (#655)

* Fix typos and minor style edits

* Fix typos and broken param name

enzocano committed:
[nodejs][20.qna-with-appinsights] Fix dependencies & readme.md (#654)

* Fix typos

* Change ABS name

* Update .bot filename

Commit:
[netcore][05.multi turn prompt] Fix typos and add improvements (#620)

* Edit readme.md

* Fix typos

* Remove Azure Storage section as per feedback

* Edit deployment step

Commit:
[netcore][03.welcome user] Fix typos and add improvements (#619)

* Edit readme.md

* Fix typos and broken param name

* Remove Azure Storage section as per feedback

* Edit deployment step

JasonSowers committed:
Merge pull request #637 from talon2king/master

Fixes to CafeBots OnTurnProperty and BookTableDialog classes

franher committed:
refactor: ES6 refactors (#678)

* refactor: 'didBotWelcomeUser' is never reassigned. Use 'const' instead

* refactor: use let instead of var

* refactor: remove unused parameter

* refactor: use let instead of var

* refactor: Delete unnecessary return await

* refactor: 'userName' is never reassigned. Use 'const' instead

myselfhimself committed:
Add missing fetch-ponyfill dependency (#648)

Otherwise `npm run start` crashes with `Error: Cannot find module 'fetch-ponyfill'`

daveta committed:
Just fix the readme this time

stevengum committed:
Merge pull request #661 from halilb/master

fixing watch script for typescript samples

JasonSowers committed:
removing bot files for samples with services (#685)

* removed bot files for c#

* removed bot files for c#

* removing bot files

* removing bot files

* removing bot files

JasonSowers committed:
Merge pull request #673 from Microsoft/daveta-publish-validate

Modify echobot and basicbot to add Publish validations

Commit:
[nodejs][01.console echo] Fix typo and add missing ampersand (#694)

* Changed single ampersand to double ampersand

* Typo in consoleAdapter.js file

Commit:
[es6][1.a.browser echo] Fix missing ampersand and udpate botbuilder-core version (#691)

* Updated botbuilder-core version

* Change ampersand in README

lucaleone committed:
Fix Startup.cs typos (#688)

fixed typos in Startup.cs comments

lucaleone committed:
removed rendundant field initializer + readonly + improved LINQ query (#687)

_isProduction has been made readonly to avoid user changes.
The linq query at line 68 has been mad shorter to improver readability.

enzocano committed:
[netcore][20.qna-with-appinsights] Fix readme.md and issue with DI (#689)

* Fix typos

* Fix dependency injection

daveta committed:
Safekeeping checkin

Commit:
Replaced single ampersand for double ampesand (#700)

Commit:
[typescript][11.qnamaker] Fix typos and improve formatting (#702)

* Fixed minor typos for README

* Fix .qna file invalid JSON format

* Fixed syntax error in .bot file

* Add missing comma in .qna file

enzocano committed:
Add steps to run the sample (#695)

enzocano committed:
[netcore][18.bot-authentication] Fix typos and Add OAuth configuration steps (#693)

* Fix typos

* Add step to configure OAuth

stevengum committed:
update the README, .luis, .lu and .qna files

daveta committed:
Merge pull request #684 from Microsoft/daveta-nodejs

Fix nodejs dispatch

stevengum committed:
fix merge conflicts

stevengum committed:
fix merge conflicts

stevengum committed:
update README.md, NlpDispatchBot.cs, CognitiveModels

stevengum committed:
update README.md, NlpDispatchBot.cs, CognitiveModels

stevengum committed:
include appsettings.json instructions in README.md

stevengum committed:
include appsettings.json instructions in README.md

stevengum committed:
add LUIS subscription key link to README

stevengum committed:
add LUIS subscription key link to README

stevengum committed:
change static readonly to consts in .netcore dispatch sample

stevengum committed:
change static readonly to consts in .netcore dispatch sample

benbrown committed:
Change ILuisService (abstract interface) to LuisService (implemented real class)
Update readme to add instructions about format .bot file
Add "future reading" link

benbrown committed:
Add a note about missing luis configuration value
Remove placeholder value for luis configuration

benbrown committed:
readme formatting tweaks

benbrown committed:
readme formatting tweaks

stevengum committed:
Merge pull request #713 from Microsoft/benbrown/fix-627

Update nlp-with-luis (fix for #627)

benbrown committed:
Update readme to include instructions to set LUIS_CONFIGURATION
Update readme with some new links to match the typescript version of this
Remove unused blob storage config from bot.recipe
Add missing 34.luis file to deploymentScripts

benbrown committed:
Update readme to include instructions to set LUIS_CONFIGURATION
Update readme with some new links to match the typescript version of this
Remove unused blob storage config from bot.recipe
Add missing 34.luis file to deploymentScripts

stevengum committed:
change public consts to private

stevengum committed:
change public consts to private

stevengum committed:
apply fixes to node basic-bot sample per #634

stevengum committed:
apply fixes to node basic-bot sample per #634

Commit:
fixing welcome messages

Commit:
fixing welcome messages

Commit:
fixing welcome messages

Commit:
fixing welcome messages

Commit:
fixing welcome messages

Commit:
fixing welcome messages

Commit:
renamed context

Commit:
renamed context

amr-elsehemy committed:
adding bot translator sample

amr-elsehemy committed:
Merge branch 'master' of https://github.com/Microsoft/BotBuilder-Samples

benbrown committed:
Update samples/javascript_nodejs/13.basic-bot/deploymentScripts/DEPLOY.md

benbrown committed:
Update samples/javascript_nodejs/13.basic-bot/deploymentScripts/DEPLOY.md

stevengum committed:
Merge pull request #717 from Microsoft/stgum/node-basic

apply fixes to node basic-bot sample per #634

stevengum committed:
Merge pull request #717 from Microsoft/stgum/node-basic

apply fixes to node basic-bot sample per #634

stevengum committed:
Merge pull request #613 from Microsoft/stgum/csharp_dispatch

 update .netcore Dispatch sample

stevengum committed:
Merge pull request #613 from Microsoft/stgum/csharp_dispatch

 update .netcore Dispatch sample

JasonSowers committed:
Merge pull request #718 from Microsoft/jassow-welcome

Update welcome messages

JasonSowers committed:
Merge pull request #718 from Microsoft/jassow-welcome

Update welcome messages

stevengum committed:
update README.md and fix index.js in nlp-with-luis (#630)

stevengum committed:
update README.md and fix index.js in nlp-with-luis (#630)

stevengum committed:
Merge pull request #715 from Microsoft/benbrown/fix-625

Update pure js version of nlp-with-luis (fix for #625)

stevengum committed:
Merge pull request #715 from Microsoft/benbrown/fix-625

Update pure js version of nlp-with-luis (fix for #625)

stevengum committed:
Merge pull request #720 from Microsoft/stgum/node-luis

update README.md and fix index.js in nlp-with-luis (#630)

stevengum committed:
Merge pull request #720 from Microsoft/stgum/node-luis

update README.md and fix index.js in nlp-with-luis (#630)

Commit:
Corrected routing in bot.js and param assignment in greeting.js

Commit:
Corrected routing in bot.js and param assignment in greeting.js

Commit:
Removed duplicated bot.js code

Commit:
Removed duplicated bot.js code

daveta committed:
Fix Recognizer to work with luisgen, fix numerous issues, enable distributed W3C correlation, use brand new App Insights

alundiak committed:
Small typo fixes in comments and string value for 'description' field in result 'package.json' file. (#722)

alundiak committed:
Small typo fixes in comments and string value for 'description' field in result 'package.json' file. (#722)

Commit:
[nodejs][03.welcome users] Fix typos and wrong parameter type, and improve formatting (#719)

* Fixed wrong .bot filenames

* Replaced deploy to azure instruction

* Changed ConversationState to UserState

* Fix typos

Commit:
[nodejs][03.welcome users] Fix typos and wrong parameter type, and improve formatting (#719)

* Fixed wrong .bot filenames

* Replaced deploy to azure instruction

* Changed ConversationState to UserState

* Fix typos

Commit:
[netcore][19.custom dialogs] Fix typos and improve formatting (#704)

* Fix typos and minor style edits

* Fix typos and minor style edit

* Move static member before non-static member

Commit:
[netcore][19.custom dialogs] Fix typos and improve formatting (#704)

* Fix typos and minor style edits

* Fix typos and minor style edit

* Move static member before non-static member

enzocano committed:
Add steps to setup blob storage (#710)

enzocano committed:
Add steps to setup blob storage (#710)

Commit:
[netcore][17.multilingual bot] Fix typos and improve formatting (#701)

* Minor style edits, edit Deploy section

* Fix typos and minor style changes

* MultiLingualBot.bot

* Fix additional typos

Commit:
[netcore][17.multilingual bot] Fix typos and improve formatting (#701)

* Minor style edits, edit Deploy section

* Fix typos and minor style changes

* MultiLingualBot.bot

* Fix additional typos

Commit:
[nodejs][04.simple prompt] Fix typos and improve formatting (#725)

* Fix typos and enhance readme structure

* Fix typos in sample scripts

* Added language to code style

Commit:
[nodejs][04.simple prompt] Fix typos and improve formatting (#725)

* Fix typos and enhance readme structure

* Fix typos in sample scripts

* Added language to code style

JasonSowers committed:
c#/20.qna-with-appinsights (#716)

* added qna

* Update 34.qna

JasonSowers committed:
c#/20.qna-with-appinsights (#716)

* added qna

* Update 34.qna

carlosscastro committed:
Message routing nodejs sample: minor fixes and improvements

carlosscastro committed:
Message routing nodejs sample: minor fixes and improvements

carlosscastro committed:
Message routing sample fixes: PR comments

carlosscastro committed:
Message routing sample fixes: PR comments

Commit:
Updates to reflect PR change requests

Commit:
Updates to reflect PR change requests

Commit:
Revert change for mistaken typo correction

stevengum committed:
Merge pull request #727 from Microsoft/ccastro/fix-multilingual-node

Multilingual-conversations Node.js sample: minor fixes and improvements

stevengum committed:
Merge pull request #727 from Microsoft/ccastro/fix-multilingual-node

Multilingual-conversations Node.js sample: minor fixes and improvements

Commit:
Added descriptor for utterance param

Commit:
Added descriptor for utterance param

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/tests/languageTranslator.test.js:
```diff
@@ -0,0 +1,123 @@
+
+    it('should translate en to fr and support html tags in sentences', async()=> {
+        
+        let translationOptions = new TranslateArrayOptions();
```
```suggestion
        const translationOptions = new TranslateArrayOptions();
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/tests/languageTranslator.test.js:
```diff
@@ -0,0 +1,123 @@
+        translationOptions.to = "fr";
+        translationOptions.texts = ["greetings <br> My friends"];
+
+        let translator = new MicrosoftTranslator(translatorKey);
```
```suggestion
        const translator = new MicrosoftTranslator(translatorKey);
```

ahmedlhanafy replied:
Do the same for all

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/tests/languageTranslator.test.js:
```diff
@@ -0,0 +1,123 @@
+        return;
+    }
+
+    it('should translate en to fr and support html tags in sentences', async()=> {
```
```suggestion
    it('should translate en to fr and support html tags in sentences', async () => {
```

ahmedlhanafy replied:
Do the same for all

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/tests/languageTranslator.test.js:
```diff
@@ -0,0 +1,123 @@
+    }
+
+    it('should translate en to fr and support html tags in sentences', async()=> {
+        
```
Remove extra line

ahmedlhanafy replied:
Do the same for all

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+"use strict";
```
Remove this

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+// Licensed under the MIT License.
+
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
```
Remove this

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+Object.defineProperty(exports, "__esModule", { value: true });
+
+import { Middleware, TurnContext, ConversationState, BotStatePropertyAccessor }  from 'botbuilder';
+import {MicrosoftTranslator, TranslateArrayOptions} from './translator';
```
```suggestion
import { MicrosoftTranslator, TranslateArrayOptions } from './translator';
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+
+import { Middleware, TurnContext, ConversationState, BotStatePropertyAccessor }  from 'botbuilder';
+import {MicrosoftTranslator, TranslateArrayOptions} from './translator';
+let _translator = null;
```
Move this inside the class

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+import {MicrosoftTranslator, TranslateArrayOptions} from './translator';
+let _translator = null;
+
+const ENGLISH_LANGUAGE = 'en';
```
Move this inside the class

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+let _translator = null;
+
+const ENGLISH_LANGUAGE = 'en';
+const DEFAULT_LANGUAGE = ENGLISH_LANGUAGE;
```
Move this inside the class

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+export class TranslatorMiddleware {
+    private translatorKey: string;
+    private to: string;
+    /**
```
Add new line

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+        @param {string} translatorKey Microsoft Text Translation API key.
+    *   @param {string} to 
+     */
+    constructor(to, translatorKey, noTranslatePatterns?: {[id: string] : string[]}, wordDictionary?: { [id: string]: string }) {
```
```suggestion
    constructor(private to, private translatorKey, noTranslatePatterns?: {[id: string] : string[]}, wordDictionary?: { [id: string]: string }) {
```
You can do this and remove declaring the instance variables manually

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+            const shouldTranslate = detectedLanguageId !== this.to;
+
+            if (shouldTranslate) {
+                var options = new TranslateArrayOptions();
```
Don't use `var`

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+                options.from = detectedLanguageId;
+                options.to = this.to;
+                options.texts = [turnContext.activity.text];    
+                turnContext.activity.text = (await _translator.translateArray(options))[0].translatedText;
```
Doing this on two different lines will make it clearer

ahmedlhanafy replied:
Otherwise people might get confused and remove the parentheses, which will break functionality

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator-middleware.ts:
```diff
@@ -0,0 +1,51 @@
+    }
+}
+
+module.exports.TranslatorMiddleware = TranslatorMiddleware;
```
You already exported the class on line:13, no need to write this. Moreover, this is more of a JS syntax try to avoid when using TS and use the alternatives

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/tsconfig.json:
```diff
@@ -0,0 +1,66 @@
+{
```
Remove comments in this file

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/typings/globals/mocha/index.d.ts:
```diff
@@ -0,0 +1,230 @@
+// Generated by typings
```
You can install this https://www.npmjs.com/package/@types/mocha as a dev dependency and remove this file

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+/**
+ * @private
+ */
+export class TranslationResult {
```
Use interfaces or types for this use case. We represent data classes in TS mostly by interfaces or types

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+import { promises } from 'fs';
+
+
+export class TranslateArrayOptions{
```
Use interfaces or types for this use case. We represent data classes in TS mostly by interfaces or types

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+/**
+ * @private
+ */
+export class TranslationRequest{
```
Use interfaces or types for this use case. We represent data classes in TS mostly by interfaces or types

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+/**
+ * @private
+ */
+export class MicrosoftTranslator {
```
Are these private or is it a typo?

amr-elsehemy replied:
it's a typo

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    apiKey: string;
+    postProcessor: TranslatorPostProcessor;
+
+    constructor(apiKey: string, noTranslatePatterns?: {[id: string] : string[]}, wordDictionary?: { [id: string]: string }) {
```
Do this 
```suggestion
    constructor(private apiKey: string, noTranslatePatterns?: {[id: string] : string[]}, wordDictionary?: { [id: string]: string }) {
```
and get rid of lines 34 and 38

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+
+    constructor(apiKey: string, noTranslatePatterns?: {[id: string] : string[]}, wordDictionary?: { [id: string]: string }) {
+        this.apiKey = apiKey;
+        this.postProcessor = new TranslatorPostProcessor();
```
Remove unnecessary initialization

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            },
+            body: JSON.stringify([translationRequest]),
+        });
+        let detectedLanguage = JSON.parse(response);
```
```suggestion
        return JSON.parse(response);
```
Unnecessary declaration

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    public detect = async (text: string): Promise<object> => {
+        let uri: any = "https://api.cognitive.microsofttranslator.com/detect?api-version=3.0";
```
```suggestion
        const uri: string = "https://api.cognitive.microsofttranslator.com/detect?api-version=3.0";
```

ahmedlhanafy replied:
```suggestion
        const uri: string = "https://api.cognitive.microsofttranslator.com/detect?api-version=3.0";
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+
+    public detect = async (text: string): Promise<object> => {
+        let uri: any = "https://api.cognitive.microsofttranslator.com/detect?api-version=3.0";
+        let translationRequest = new TranslationRequest();
```
```suggestion
        const translationRequest = new TranslationRequest();
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+ * @private
+ */
+export class TranslationRequest{
+	Text: string;
```
```suggestion
	text: string;
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        return detectedLanguage;
+    };
+
+    public translateArray = async (options: TranslateArrayOptions):  Promise<object> => {
```
```suggestion
    public translateArray = async (options: TranslateArrayOptions): Promise<object> => {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    };
+
+    public translateArray = async (options: TranslateArrayOptions):  Promise<object> => {
+		try {
```
Fix indentation

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+
+    public translateArray = async (options: TranslateArrayOptions):  Promise<object> => {
+		try {
+            let from = options.from;
```
```suggestion
            const { from, to, texts } = options;
```
instead of lines 63, 64 and 65

amr-elsehemy replied:
texts is not a const

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            let results = [];
+            texts.forEach((text, index, array) => {
+                let normalizedText = text.trim();
+                    orgTexts.push(normalizedText);
```
Fix indentation

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            let orgTexts = [];
+            let translationRequests = [];
+            let results = [];
+            texts.forEach((text, index, array) => {
```
```suggestion
            texts.forEach((text, index) => {
```
Remove unused variable

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                    translationRequest.Text = normalizedText;
+                    translationRequests.push(translationRequest);
+            });
+            let uri: any = `https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&includeAlignment=true&includeSentenceLength=true&from=${from}&to=${to}`;
```
```suggestion
            const uri: string = `https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&includeAlignment=true&includeSentenceLength=true&from=${from}&to=${to}`;
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+				url: uri,
+				method: 'POST',
+				headers: {
+                    'Ocp-Apim-Subscription-Key': this.apiKey,
```
Fix indentation

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+				method: 'POST',
+				headers: {
+                    'Ocp-Apim-Subscription-Key': this.apiKey,
+					'Content-Type': 'application/json',
```
Fix indentation

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+				},
+				body: JSON.stringify(translationRequests),
+            });
+            let translations = JSON.parse(response);
```
```suggestion
            const translations = JSON.parse(response);
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+				body: JSON.stringify(translationRequests),
+            });
+            let translations = JSON.parse(response);
+            translations.forEach((traslationModel, index, array) => {
```
```suggestion
            translations.forEach((traslationModel, index) => {
```
Remove unused variable

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            });
+            let translations = JSON.parse(response);
+            translations.forEach((traslationModel, index, array) => {
+                                let translation = traslationModel.translations[0].text;
```
```suggestion
                                const translation = traslationModel.translations[0].text;
```

amr-elsehemy replied:
it's not a const

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                                    let alignment = traslationModel.translations[0].alignment.proj;
+                                    translation = this.postProcessor.fixTranslation(orgTexts[index], alignment, translation, from);
+                                }
+                                let result: TranslationResult = { translatedText: translation };
```
```suggestion
                                results.push({ translatedText: translation })
```
And remove line 95

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+
+
+/**
+ * @private
```
Are you sure that this class is private or is it just a typo?

amr-elsehemy replied:
it's private

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        }
+
+        if(noTranslatePatterns) {
+            for(var key in noTranslatePatterns) {
```
```suggestion
            for(const key in noTranslatePatterns) {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        
+        if (wordDictionary) {
+            Object.keys(this.wordDictionary).forEach(word => {
+                if (word != word.toLowerCase()) {
```
```suggestion
                if (word !== word.toLowerCase()) {
```
Use strict equality

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+
+        if(noTranslatePatterns) {
+            for(var key in noTranslatePatterns) {
+                noTranslatePatterns[key].forEach((pattern, index, array) => {
```
```suggestion
                noTranslatePatterns[key].forEach((pattern, index) => {
```
Remove unused variable

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                    if(!(key in this.noTranslatePatterns)){
+                        this.noTranslatePatterns[key] = [];
+                    }
+                    if (pattern.indexOf('(') == -1) {
```
```suggestion
                    if (pattern.indexOf('(') === -1) {
```
Use strict equality

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    private splitSentence(sentence: string, alignments: string[], isSrcSentence = true): string[] {
+        let wrds = sentence.split(' ');
```
```suggestion
        const words = sentence.split(' ');
```
1) Use `const`
2) Use clear variable names

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+
+    private splitSentence(sentence: string, alignments: string[], isSrcSentence = true): string[] {
+        let wrds = sentence.split(' ');
+        let lastWrd = wrds[wrds.length - 1];
```
```suggestion
        const lastWord = wrds[wrds.length - 1];
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    private splitSentence(sentence: string, alignments: string[], isSrcSentence = true): string[] {
+        let wrds = sentence.split(' ');
+        let lastWrd = wrds[wrds.length - 1];
+        if (".,:;?!".indexOf(lastWrd[lastWrd.length - 1]) != -1) {
```
```suggestion
        if (".,:;?!".indexOf(lastWrd[lastWrd.length - 1]) !== -1) {
```
Use strict equality

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        if (".,:;?!".indexOf(lastWrd[lastWrd.length - 1]) != -1) {
+            wrds[wrds.length - 1] = lastWrd.substr(0, lastWrd.length - 1);
+        }
+        let alignSplitWrds: string[] = [];
```
Use `const`

amr-elsehemy replied:
alignSplitWrds is gonna be re-assigned

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            wrds[wrds.length - 1] = lastWrd.substr(0, lastWrd.length - 1);
+        }
+        let alignSplitWrds: string[] = [];
+        let outWrds: string[] = [];
```
Use const

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        }
+        let alignSplitWrds: string[] = [];
+        let outWrds: string[] = [];
+        let wrdIndexInAlignment = 1;
```
```suggestion
        const wordIndexInAlignment = 1;
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            wrdIndexInAlignment = 0;
+        } else {
+            alignments.sort((a, b) => {
+                let aIndex = parseInt(a.split('-')[wrdIndexInAlignment].split(':')[0]);
```
Use const

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        } else {
+            alignments.sort((a, b) => {
+                let aIndex = parseInt(a.split('-')[wrdIndexInAlignment].split(':')[0]);
+                let bIndex = parseInt(b.split('-')[wrdIndexInAlignment].split(':')[0]);
```
Use const

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                }
+            });
+        }
+        let sentenceWithoutSpaces = sentence.replace(/\s/g, '');
```
Use `const` in lines 175 to 182

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            }
+            newWrds[outWrds.length] = wrd;
+            
+            let subSentence = this.join("", newWrds);
```
Use `const`

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    private keepSrcWrdInTranslation(alignment: { [id: number] : number }, sourceWords: string[], targetWords: string[], srcWrdIndex: number) {
+        if (!(typeof alignment[srcWrdIndex] === "undefined")) {
```
```suggestion
        if (alignment[srcWrdIndex] !== undefined)) {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    private replaceWordInDictionary(alignment: { [id: number] : number }, sourceWords: string[], targetWords: string[], srcWrdIndex: number) {
+        if (!(typeof alignment[srcWrdIndex] === "undefined")) {
```
```suggestion
        if (alignment[srcWrdIndex] !== undefined)) {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        return targetWords;
+    }
+
+    public fixTranslation(sourceMessage: string, alignment: string, targetMessage: string, languageId : string): string {
```
Add comment to public methods

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        let numericMatches = sourceMessage.match(/[0-9]+/g);
+        let containsNum = numericMatches != null;
+        
+        if ((!containsNum && (!this.noTranslatePatterns || !this.noTranslatePatterns[languageId] || this.noTranslatePatterns[languageId].length == 0) && !this.wordDictionary) || alignment.trim() == '') {
```
```suggestion
        if ((!containsNum && (!this.noTranslatePatterns || !this.noTranslatePatterns[languageId] || this.noTranslatePatterns[languageId].length == 0) && !this.wordDictionary) || alignment.trim() === '') {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+            this.noTranslatePatterns[languageId].forEach(pattern => {
+                let regExp = new RegExp(pattern, "i");
+                let matches = sourceMessage.match(regExp);
+                if (matches != null) {
```
```suggestion
                if (matches !== null) {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                            return;
+                        }
+                        newNoTranslateArrayLength++;
+                        // newChrLengthFromMatch += srcWords[wrdIndx].length + 1;
```
Remove comment

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                        newChrLengthFromMatch += srcWords[wrdIndx].length;
+                    }
+                    if (chrIndx + wrd.length < sourceMessageCharacters.length && sourceMessageCharacters[chrIndx + wrd.length] == ' ')
+                    {
```
curly bracket should be in the same line with the `if` statement

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                    {
+                        chrIndx += wrd.length + 1;
+                    }
+                    else{
```
```suggestion
                    else {
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                    wrdIndx++;
+
+                    if (srcIndx == -1)
+                    {
```
curly bracket should be in the same line with the `if` statement

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                        // newChrLengthFromMatch += srcWords[wrdIndx].length + 1;
+                        newChrLengthFromMatch += srcWords[wrdIndx].length;
+                    }
+                    if (chrIndx + wrd.length < sourceMessageCharacters.length && sourceMessageCharacters[chrIndx + wrd.length] == ' ')
```
```suggestion
                    if (chrIndx + wrd.length < sourceMessageCharacters.length && sourceMessageCharacters[chrIndx + wrd.length] === ' ')
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+                    }
+                    wrdIndx++;
+
+                    if (srcIndx == -1)
```
```suggestion
                    if (srcIndx === -1)
```

ahmedlhanafy commented on samples/javascript_typescript/60.multilingual-luis-bot/translator.ts:
```diff
@@ -0,0 +1,345 @@
+        }
+        
+        if(this.wordDictionary){
+            srcWords.forEach((wrd, index, Array) => {
```
```suggestion
            srcWords.forEach((wrd, index) => {
```

ahmedlhanafy reviewed (changes requested):
General comments: 
1) Most of `let` can be replaced with `const`, since most of them are not being reassigned to other variables
2) Always use strict equality https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

stevengum committed:
Merge pull request #721 from Microsoft/vstkanb-message-routing

vstkanb message routing

stevengum committed:
Merge pull request #721 from Microsoft/vstkanb-message-routing

vstkanb message routing

lucaleone committed:
Added WelcomeUserState + fix Exception message + improved Linq query (#696)

WelcomeUser project:
Implemented same architecture of the EchoBotWithCounter project. The project now contains a WelcomeUserState class handled by WelcomeUserStateAccessors.

line 62 all Startup.cs
botConfig in the InvalidOperationException message is always null, it would be better for the user to provide the path in which the .bot file shold be.

Line 66 all Startup.cs
Improved LINQ query

lucaleone committed:
Added WelcomeUserState + fix Exception message + improved Linq query (#696)

WelcomeUser project:
Implemented same architecture of the EchoBotWithCounter project. The project now contains a WelcomeUserState class handled by WelcomeUserStateAccessors.

line 62 all Startup.cs
botConfig in the InvalidOperationException message is always null, it would be better for the user to provide the path in which the .bot file shold be.

Line 66 all Startup.cs
Improved LINQ query

amr-elsehemy committed:
resolving PR comments

amr-elsehemy committed:
Merge branch 'master' of https://github.com/Microsoft/BotBuilder-Samples

amr-elsehemy committed:
- enriching readme
- re-structure directories

Commit:
Corrected wrong comment correction

Commit:
Minor typo in cafeLocation list

Commit:
Minor typo for parameter description

daveta committed:
Merge pull request #723 from Microsoft/daveta-enterprise

Update Enterprise to add improved AppInsights

JasonSowers committed:
SPEC update

JasonSowers committed:
Update SPEC.md

VictorGrycuk committed:
Fix validation for no answers found for QnA

VictorGrycuk committed:
Fix broken links and minor style edits

VictorGrycuk committed:
Replace string concatenation with string template

VictorGrycuk committed:
Fix typos and minor style edits to .cs files

VictorGrycuk committed:
Replace accent character for single quote

VictorGrycuk committed:
Remove unnecessary calling

VictorGrycuk committed:
Minor style edits to resource files

daveta committed:
Merge pull request #755 from southworkscom/southworks/netcore/cafe-bot/qna-results-validation

[netcore][51.bot cafe] Fix QnA results validation

daveta committed:
Merge pull request #753 from southworkscom/southworks/netcore/cafe-bot/readme

[netcore][51.bot cafe] Fix broken links and various style edits in readme.md

daveta committed:
Merge pull request #750 from southworkscom/southworks/netcore/cafe-bot/use-string-template

[netcore][51.cafe-bot] Replace string concatenation with string template

daveta committed:
Merge pull request #746 from southworkscom/southworks/netcore/cafe-bot/resource-style-edits

[netcore][51.cafe bot] Fix typos in Resources files

daveta committed:
Merge pull request #747 from southworkscom/southworks/netcore/cafe-bot/code-style-edits

[netcore][51.cafe bot] Fix typos and improve formatting in .cs files

daveta committed:
Merge pull request #751 from southworkscom/southworks/nodejs/cafe-bot

[nodejs][51.cafe bot] Fix typos and improve formatting

Commit:
[netcore][10.prompt validations] Fix typos and improve formatting (#769)

* Style tweaks in readme

* Minor style edits in code

Commit:
[netcore][12.nlp with luis] Fix typos and improve formatting (#771)

* Style edits to readme

* Minor style tweaks in code

* Fix Deploy to Azure section

Commit:
[netcore][08.suggested actions] Fix typos and improve formatting (#770)

* Minor style edits in code

* Style tweaks in readme.md

* Fix Typo

Commit:
Minor typos fixes (#762)

Commit:
Fixed typos (#761)

Commit:
[netcore][23.facebook events] Improve text formatting and content (#745)

* Minor style edits

* Minor style tweaks in readme.md

Commit:
Fixed typos and minor structural changes (#741)

Commit:
Fixed typos and minor structural changes (#743)

Commit:
Fix typos (#740)

Commit:
Fixed typos and minor structural changes (#736)

Commit:
[nodejs][07.using adaptive cards] Improve formatting (#735)

* Fixed typos and minor structural changes

* Added language to code block

Commit:
Fix typos and enhance readme structure (#732)

Commit:
Fixed typos and minor structural changes (#734)

Commit:
[typescript][12.nlp with luis] Fix typos and improve Readme (#664)

* Fixed minor typos for README

* Missed changes and extra information added

* Removed old example

* Changed 'Deploy to Azure' section

* Enhance deployment information

JasonSowers committed:
Merge pull request #764 from Microsoft/JasonSowers-spec

Update SPEC.md file to reflect intended changes for README.md files

Commit:
Fix minor typos and structural changes (#777)

Commit:
[netcore][14.nlp with dispatch] Fix typos and improve formatting (#776)

* Fix broken links and tweak text style

* Fix typos and minor style tweaks

* Minor style and typo edits

* Add modifications as per feedback

lauren-mills committed:
Lamil/removing enterprisebot (#780)

* removed enterprise bot projects from samples

* updated readme with new links to enterprise bot

Commit:
[webapi][13.basic bot] Fix typos and improve formatting (#784)

* README structure enhancement and typos fixes

* Fix minor typo

* Fix minor typo in BotConfig.cs

* Minor typo fix in dialog text

Commit:
[nodejs][09.message routing] Add deploying section (#787)

* Changed steps to try sample

* Missed style changes and URL reference

* Fixed minor typos

* Removed unnecessary LUIS section

* Remove repeated word

Commit:
Missed typos and wrong path for sample folder in README (#786)

Commit:
Fix string template in QnADialog.cs (#749)

sbuffkin committed:
Add preliminary implemenation of check for if .bot file exists (#765)

Commit:
[netcore][51.cafe bot] Add missing QnA resource files (#785)

* Add missing files

* fix typo in qna.lu

amr-elsehemy committed:
- Merge branch 'master' of https://github.com/Microsoft/BotBuilder-Samples
- Resolve merge conflicts
